clarify DST transition for cron (#952)
temporalio/documentation: docs/concepts/what-is-a-temporal-cron-job.md

If you need to use time zones, here are a few edge cases to keep in mind:

> - **Beware Daylight Savings**: If a Temporal Cron Job is scheduled around the time when daylight saving time begins or ends, e.g. `30 2 * * *`, **it may run zero, one, or two times in a day**! The Cron library that we use does not do any special handling of DST transitions. Avoid schedules that include times that fall within DST transition periods.

Reviewer: ```suggestion
- **Beware Daylight Saving Time**: If a Temporal Cron Job is scheduled around the time when daylight saving time (DST) begins or ends (for example, `30 2 * * *`), **it might run zero, one, or two times in a day**! The Cron library that we use does not do any special handling of DST transitions. Avoid schedules that include times that fall within DST transition periods.
```

>   - For example, in the US, DST is at the 1AM hour. When you "fall back", the clock will go `1:59 … 1:00 … 1:01 … 1:59 … 2:00 … 2:01 AM` and any Cron jobs that falls in that 1AM hour will be fired again. The inverse happens when clocks "spring forward" for DST, and Cron jobs that fall in the 1AM hour are skipped.

Reviewer: ```suggestion
  - For example, in the US, DST begins at 2 AM. When you "fall back," the clock goes `1:59 … 1:00 … 1:01 … 1:59 … 2:00 … 2:01 AM` and any Cron jobs that fall in that 1 AM hour are fired again. The inverse happens when clocks "spring forward" for DST, and Cron jobs that fall in the 2 AM hour are skipped.
```

>   - In other timezones like Chile and Iran, DST "spring forward" is at midnight. [11:59PM is followed by 1:00AM, which means `00:00:00` never happens](https://twitter.com/raphaelschaad/status/1425249265643839490?s=20&t=vrMx4ixqcBRp5D7xB4FgNw).

Reviewer: ```suggestion
  - In other time zones like Chile and Iran, DST "spring forward" is at midnight. [11:59 PM is followed by 1 AM, which means `00:00:00` never happens](https://twitter.com/raphaelschaad/status/1425249265643839490?s=20&t=vrMx4ixqcBRp5D7xB4FgNw).
```

- **Self Hosting note**: If you manage your own Temporal Cluster, you are responsible for ensuring that it has access to current `tzdata` files. The official Docker images are built with [tzdata](https://docs.w3cub.com/go/time/tzdata/index) installed (provided by Alpine Linux), but ultimately you should be aware of how tzdata is deployed and updated in your infrastucture.
- **Updating Temporal**: If you use the official Docker images, note that an upgrade of the Temporal Cluster may include an update to the tzdata files, which may change the meaning of your Cron Schedule. You should be aware of upcoming changes to the definitions of the time zones you use, particularly around daylight saving time start/end dates.

> - **Absolute Time Fixed at Start**: The absolute start time of the next Run is computed and stored in the database when the previous Run completes, and is not recomputed. This means that if you have a Cron Schedule that runs very infrequently, and the definition of the time zone changes in between one Run and the next, the Run might happen at the wrong time. For example, `CRON_TZ=America/Los_Angeles 0 12 11 11 *` means "noon in Los Angeles on November 11" (normally not in DST). If at some point the government makes any changes (e.g. move the end of DST one week later, or stay on permanent DST year-round) the meaning of that specification will change. In that first year, the Run will happen at the wrong time, because it was computed using the older definition.

Reviewer: ```suggestion
- **Absolute Time Fixed at Start**: The absolute start time of the next Run is computed and stored in the database when the previous Run completes, and is not recomputed. This means that if you have a Cron Schedule that runs very infrequently, and the definition of the time zone changes between one Run and the next, the Run might happen at the wrong time. For example, `CRON_TZ=America/Los_Angeles 0 12 11 11 *` means "noon in Los Angeles on November 11" (normally not in DST). If at some point the government makes any changes (for example, move the end of DST one week later, or stay on permanent DST year-round), the meaning of that specification changes. In that first year, the Run happens at the wrong time, because it was computed using the older definition.
```
